Model/Method/AbstractMethod.php

```diff
@@ -823,17 +823,17 @@ protected function getFailureMessage($response)
     }
 
     /**
+     * @param $transactionResponse
+     * @param $errorType
      * @return bool
      */
-    public function hasError($transactionResponse): bool
+    public function hasError($transactionResponse, $errorType): bool
     {
-        return !empty($transactionResponse->RequestErrors) && (
-                !empty($transactionResponse->RequestErrors->ChannelError) ||
-                !empty($transactionResponse->RequestErrors->ServiceError) ||
-                !empty($transactionResponse->RequestErrors->ActionError) ||
-                !empty($transactionResponse->RequestErrors->ParameterError) ||
-                !empty($transactionResponse->RequestErrors->CustomParameterError)
-            );
+        if(!empty($transactionResponse->RequestErrors) && !empty($transactionResponse->RequestErrors->$errorType)) {
```

Comment:
why not do `return !empty($transactionResponse->RequestErrors) && !empty($transactionResponse->RequestErrors->$errorType)` ?

Reply:
good question :)